embedded/sql/engine.go

```diff
@@ -113,29 +113,31 @@ func NewEngine(catalogStore, dataStore *store.ImmuStore, prefix []byte) (*Engine
 
 	copy(e.prefix, prefix)
 
-	err := e.LoadCatalog()
-	if err != nil {
-		return nil, err
-	}
-
 	return e, nil
 }
 
 // TODO (jeroiraz); this operation won't be needed with a transactional in-memory catalog
-func (e *Engine) EnsureCatalogReady() error {
+func (e *Engine) EnsureCatalogReady(cancellation <-chan struct{}) error {
```

Comment:
Not necessarily for this PR but I think we should start passing [`context.Context`](https://pkg.go.dev/context#Context) around - it seems to be the native way to send cancellation requests in golang.

Reply:
sure